### ISSUE TITLE
Is there any support for Dual-Core Skylake Processors?

### ISSUE THREAD

**Issue by urbanman2004** [Piker-Alpha/ssdtPRGen.sh#233]
I attempted to create a SSDT file by editing the Haswell.cfg to include necessary information applicable to my CPU. I am using G4500 and I have included a copy of my cfg file as an attachment as well. I was following the instructions on this guide [here](http://www.insanelymac.com/forum/topic/303755-how-to-intel-pentium-g3xxx-and-celeron-g1xxx-haswell-works-with-speedstep-yosemite-1010-and-mavericks-109/) thinking its method could possibly work for my CPU as well. I keep getting error: "Error: Invalid argument detected: -h (check ssdtPRGen.sh -h) Aborting ..." Any assistance is greatly appreciated.
Here are my following specs:

Intel Pentium Processor G4500
ASUS Z170M-PLUS Micro ATX Motherboard
Mushkin Enhanced ECO2 512GB SSD
Corsair Vengeance LPX 16GB (2x8GB) DDR4
HGST Travelstar 7K1000 1TB 2.5" 7200RPM HDD
Seagate Barracuda 3TB 3.5" 7200RPM HDD ST3000DM001
LEPA AquaChanger 240 Liquid CPU Cooler 
Corsair Carbide 88R Mid-Tower Case
LG Electronics 14x SATA Blu-ray Rewriter
PNY GTX 960 XLR8 4GB
CORSAIR CX750M 750W PSU

[Haswell.cfg.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/340863/Haswell.cfg.zip)

**Comment by Piker-Alpha** [comment]
Why? The Intel G4500 processor is [already supported](https://github.com/Piker-Alpha/ssdtPRGen.sh/blob/Beta/Data/Skylake.cfg#L85).

**Comment by urbanman2004** [comment]
O ok, disregard. How long has it been supported? I could've sworn when I was trying to build an SSDT back in April through the automated method by having it prepopulate the settings in Terminal I would always get that same error message.

**Comment by Piker-Alpha** [comment]
Since [Sep 2, 2015](https://github.com/Piker-Alpha/ssdtPRGen.sh/blame/Beta/Data/Skylake.cfg#L85)

**Comment by Piker-Alpha** [comment]
Can we close this issue?

**Comment by urbanman2004** [comment]
Sorry for the delay, fell asleep. I still havent got a chance to apply the script which u say supports Skylake. My question is how come the Skylake.cfg you presented as a reference [here](https://github.com/Piker-Alpha/ssdtPRGen.sh/blame/Beta/Data/Skylake.cfg#L85) isnt prepopulating in the _~/library/ssdtPRGen/Data_ folder?

**Comment by Piker-Alpha** [comment]
You are using the beta version of the script?

Please verify that your copy of ~/Library/ssdtPRGen/Data/Skylake.cfg includes the processor data for your processor If not, then trash the file and launch the script (file will be downloaded).

**Comment by urbanman2004** [comment]
K. I will try to get it working with the normal version, not the Beta and share my findings/results.

**Comment by urbanman2004** [comment]
So I been using the link to the Beta [here](curl -o ~/ssdtPRGen.sh https://raw.githubusercontent.com/Piker-Alpha/ssdtPRGen.sh/Beta/ssdtPRGen.sh) which I'm not sure is the latest one since that's the only one I could find. For some reason it wont download the 'Data' folder in order to prepopulate the Skylake.cfg file. I have attached a copy of the Terminal readout file from my session. Hopefully it could be of some use to point out what could be the cause to my problem.
[Terminal Saved Output.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/342793/Terminal.Saved.Output.zip)

**Comment by Piker-Alpha** [comment]
Just launch the script _without_ any arguments and let it download all the data that you need. 
Don't use unsupported arguments. See also: `./ssdtPRGen.sh -help`

**Comment by urbanman2004** [comment]
O ok... I just got off work. I will try your suggestion in a couple of hours

**Comment by urbanman2004** [comment]
So I did as you suggested and I didnt use unsupported arguments but I still cant build the SSDT. Here  is my attached copy of the Terminal readout file from my session.
[Terminal Saved Output.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/344741/Terminal.Saved.Output.zip)

**Comment by Piker-Alpha** [comment]
You also need to remove previously generated and downloaded file with:
`sudo rm -R ~/Library/ssdtPRGen`
After that you can run the freshly downloaded script.

**Comment by urbanman2004** [comment]
Thanks for the advice man, I got it to work and was finally able to create a system specific SSDT.aml. I attached a copy of the Terminal readout file from my session. Is there anything I should be on the look out for or am I good? Thanks again
[Terminal Saved Output.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/345270/Terminal.Saved.Output.zip)

**Comment by Piker-Alpha** [comment]
You don't need `-cpus 1 -target 4` and if you really want to use a Haswell based SMBIOS then everything is fine.

**Comment by urbanman2004** [comment]
O ok... I rather not still use a Haswell based SMBIOS but Idk of any sys def that are compatible w/ Skylake. If I remember correctly, I tried using iMac17,2 (only Skylake sys def I know of that Apple has created) some time ago but it would conflict with my graphics, ~~causing my display to only show 1072x768 as the only resolution available~~. I also attached a copy of the Terminal readout file from my session.
[Terminal Saved Output.zip](https://github.com/Piker-Alpha/ssdtPRGen.sh/files/345346/Terminal.Saved.Output.zip)
---EDIT---
Correction, using any sys def other than iMac14,2 causes my screen to go blank (No signal) upon startup.
